lib/Cake/Network/Email/CakeEmail.php

```diff
@@ -1717,6 +1721,27 @@ protected function _renderTemplates($content) {
 			$rendered[$type] = implode("\n", $rendered[$type]);
 			$rendered[$type] = rtrim($rendered[$type], "\n");
 		}
+
+		/* Embed images inline in html templates */
+		if (!empty($rendered['html'])) {
```

Comment:
Only renderTemplate:
```
$loops = 10000;
$t1 = explode(' ', microtime());
for ($i = 0; $i < $loops; $i++) {
			$rendered = parent::_renderTemplates($content);
}
$t2 = explode(' ', microtime());
$t = $t2[1] - $t1[1];
$t += $t2[0] - $t1[0];
debug("renderTemplates:\t" . $t);

```
renderTemplates:      46.175512 seconds "only" 10 000 loops!!!
In loops of preg_match_all php are over 12Mb... in this loop are over 23Mb...
Any template system is, by definition inefficient in terms of performance.